dima/react-movies/src/pages/mainpage/mainpage.tsx

```diff
@@ -0,0 +1,199 @@
+import SortFilter from "../../components/sortFilter";
+import Footer from "../../components/footer";
+import * as QueryString from "query-string";
+import { RouteComponentProps } from "react-router-dom";
+
+import "./mainpage.css";
+import SortProperty from "../../enums/SortProperty";
+import FilterProperty from "../../enums/FilterPropery";
+import ParamsToPush from "../../interfaces/paramsToPush";
+import { MainConnectProps } from ".";
+
+interface OwnProps {
+  currentSortType: string;
+  setCurrentSortType: (currentSortType: SortProperty) => void;
+}
+
+type MainPageProps = OwnProps &
+  MainConnectProps &
+  RouteComponentProps<{ searchBy: FilterProperty }>;
+
+class MainPage extends React.Component<
+  MainPageProps & RouteComponentProps<{ searchBy: string }>
+> {
+  componentDidMount() {
+    const params = QueryString.parse(this.props.history.location.search);
+    let pageNum = params.page ? Number(params.page.toString()) - 1 : 0;
+    this.fetchMovies(pageNum);
+  }
+
+  componentDidUpdate(
+    prevProps: MainPageProps & RouteComponentProps & MainConnectProps
+  ) {
+    if (prevProps.location.search !== this.props.location.search) {
+      const params = QueryString.parse(this.props.history.location.search);
+      let pageNum = params.page ? Number(params.page.toString()) - 1 : 0;
+      this.fetchMovies(pageNum);
+    }
+  }
+  componentWillUnmount(): void {
+    this.props.resetMovies();
+  }
+
+  pushParams = (urlParams: ParamsToPush): void => {
+    const { history } = this.props;
+    history.push({
+      pathname: "/search",
+      search: QueryString.stringify({ ...urlParams, page: 1 }),
+    });
+  };
+
+  compareSortFromUrlToState = (sortBy: string | string[] | null) =>
+    sortBy !== this.props.currentSortType;
+
+  fetchMovies = (pageNum: number): void => {
+    let defaultParams = {
+      limit: 9,
+      sortBy: "release_date",
+      sortOrder: "desc",
+      offset: 0,
+    };
+    const { location, currentSortType } = this.props;
+    const { date, rating } = SortProperty;
+    const oldParamsObj = QueryString.parse(location.search);
+    const { sortBy } = oldParamsObj;
+
+    switch (location.pathname) {
+      case "/":
+        if (sortBy && this.compareSortFromUrlToState(sortBy))
+          this.props.setCurrentSortType(sortBy === date ? date : rating);
+
+        this.props.fetchMovies({
+          ...defaultParams,
+          sortBy: oldParamsObj.sortBy ? oldParamsObj.sortBy : currentSortType,
+          offset: pageNum * 9,
+        });
+
+        break;
+
+      case "/search":
+        if (sortBy && this.compareSortFromUrlToState(sortBy))
+          this.props.setCurrentSortType(sortBy === date ? date : rating);
+        this.props.fetchMovies({
+          ...defaultParams,
+          ...oldParamsObj,
+          sortBy: oldParamsObj.sortBy ? oldParamsObj.sortBy : currentSortType,
+          offset: pageNum * 9,
+        });
+        break;
+    }
+  };
```

Comment:
redundant switch case! You actually don't need it. If search is empty, ** oldParamsObj** will be an empty object. so you can spred it and nothing will be changed